### ISSUE TITLE
Proposal: Remove column from AST and pickles

### ISSUE THREAD

**Issue by aslakhellesoy** [cucumber/cucumber#751]
Computing the column for certain AST nodes (table cells) is tricky to do consistently on all platforms. It becomes tricky when the cells have unicode characters that use several bytes such as emoji (and I think, certain asian symbols).

To work around this I have resorted to things like [this](https://github.com/cucumber/cucumber/blob/107d8db30871edc5cf175a08986a14c9ad50cc2e/gherkin/java/src/main/java/io/cucumber/gherkin/StringUtils.java#L35-L77)

Even if this might now be working correctly in Java I am pretty sure other implementations are not behaving the same (we just don't have tests to prove it).

Rather than trying to implement this correctly everywhere I want to challenge whether we need the column at all. If we don't, we can remove a fair amount of complicated/buggy code.

My memory is a little fuzzy on why we added column. I think @gasparnagy added it first since he wrote the first Gherkin implementation of the current architecture (based on his Berp parser generator).

My guess is that we thought it might come in handy for displaying results and errors inside an editor. I'm not sure if anything uses the `column` field though.

And what is a `column` anyway? Here is something you can try:

* Open an editor
* Create a new blank document
* Paste in these three monkeys on the first line: 🙈🙉🙊
* Place the cursor at the end of the line

Most editors use 1-based indexing of columns, so if you place your cursor after the last monkey, you'd expect to be on column 4. IntelliJ does this. Atom however, reports column 7!

Both are correct - they just use different ways of counting the offset:
- IntelliJ uses the [code point]() position
- Atom uses the number of UTF-16 character position, and emojis are stored as *surrogate pairs* - two characters.

Since we cannot rely on editors to use the `column` consistently, and since I have a suspicion nothing is using it at all, I propose we remove it altogether.

WDYT @vincent-psarga @charlierudolph @mpkorstanje @gasparnagy @SabotageAndi and others. Will anything break if we do?

**Comment by mpkorstanje** [comment]
When I hook everything up right with JUnit 5 IDEA will automatically jump the editor to line and column in a feature file. It is a really nice user experience. So I would like to keep it. 

And now that we are using Java 8 you can remove most of the code `StringUtils`. So I don't think it's worth removing. 

https://github.com/cucumber/cucumber/commit/dd2d62678a4382b02bcf79773f524d46fdea7023

I would use the code point position. Visually it makes most sense and it is in line with what the Unicode consortium considers a character.

**Comment by gasparnagy** [comment]
TL;DR: SpecFlow and co. uses it, but not business critical. 

I don't think it is super crucial, but both SpecFlow and the Deveroom Visual Studio extension uses that. SpecFlow for providing more targeted error message, Deveroom uses that for table formatting and syntax coloring. Having slightly improper column value for these special cases, like emojis does not cause much harm and for many languages it works fine. (Supporting special character languages is a different topic, but for those the entire concept of pipe-formatted table somewhat breaks, because their fonts are not proportional.)

Alternatively we can keep the column in the AST, but use 0 as column value for the cases when column is "unknown" for those tools where it would be too complicated to calculate the column value. (This is a kind of usual pattern anyway.)

**Comment by aslakhellesoy** [comment]
Ok, we'll keep it
